### DISCUSSION TITLE
golangci-lint action: exit code on warning severity

### DISCUSSION THREAD

**Discussion by katexochen** [golangci/golangci-lint#2639] [Q&A]
The GitHub action fails even if only warnings are reported. Is it possible to configure the action in a way that is fails on error but doesn't fail when only warnings are thrown?

**Comment by bombsimon** [comment]
I suspect this is the same as #2577 and #2523 which provides some information and context. Maybe one of the suggested alternatives would work if you still want to ignore the fact that some linters fail?

**Reply by katexochen** [reply]
As far as I understand these discussions, you can _either_ exit the run with status code 1 if something was found (regardless of the severity) _or_ always exit with status code 0 (using a workaround). Or did I miss something?
I'd like to have a status code of 1 if there are findings with severity error but a status code of 0 if there are only findings with severity warning. So the status code should depend on the severity of the issues that were found (because why would you want an CI job to fail if the severity of issues was only info/warning?).
But based on those discussions, it doesn't sound like this is possible.

By the way, thank you so much @bombsimon for great, continuous and fast support! :heart:

**Reply by bombsimon** [reply]
Ah, my bad! I guess you mean reported issues from a linter if the severity is configured as a warning? I actually don't know too much about that feature and how/if the exit code is related to the action itself. I need to look into this to give a better answer. Personally I don't use any severity configuration and disables/ignores linters and warnings that I don't see as failures. 

Feel free to share if you have a reproducible example.

**Reply by ldez** [reply]
the `severity` is not related to the exit code. It's just information.

**Reply by katexochen** [reply]
The only workaround I can currently think of is to use two different actions, one configured with the linters that only should warn and exit code 0, the other one configured for error and exit code 1. But this would lead to quite an overhead, since you would have to manage multiple config files in one repo, and users would need to manually run the different configs one at a time (outside of the action).

I guess adding this as a feature isn't desired from your side due to performance? I looked a bit into the code, one would have to iterate over all found issues to determine the error code... Another problem might be that the severity isn't an enum but a string, because of syntax difference between distinct providers. Also, this would ideally only affect the action, not golanci-lint itself, but I'm not familiar with GitHub actions nor Typescript, so I'm unsure if this could be solved on this side.

I think the rules of many linters are very important in general but have to many edge cases or false positives to let CI fail. The reviewer should decide on those issues, so it would be really nice if they were shown in the PR as warnings, instead of just disabling them.

> Feel free to share if you have a reproducible example.

```yaml
linters:
  enable:
    - deadcode
    - errcheck

severity:
  default-severity: error
  rules:
    - linters:
        - errcheck
      severity: warning
```
```go
package main

func main() {
	close() // errcheck -> warning -> exit code 0
}

func close() error { return nil }

// uncomment, deadcode -> error -> exit code 1
// func bar() {}
```
